content/switching-to-v0-mangling-on-nightly.md

```diff
@@ -0,0 +1,190 @@
+closures and generic functions.
+
+It's easy to see the new mangling scheme in action, consider the following
+example:
+
+```rust
+fn foo<T>() {
+    panic!()
+}
+
+fn main() {
+    foo::<Vec<(String, &[u8; 123])>>();
+}
+```
+
+With the legacy mangling scheme, all of the useful information about the generic
+instantiation of `foo` is lost in the symbol `f::foo`..
+
+```
+thread 'main' panicked at f.rs:2:5:
+explicit panic
+stack backtrace:
+  0: std::panicking::begin_panic
+    at /rustc/d6c...582/library/std/src/panicking.rs:769:5
+  1: f::foo
+  2: f::main
+  3: core::ops::function::FnOnce::call_once
+note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
+```
+
+..but with the v0 mangling scheme, the useful details of the generic instantiation
+are preserved with  `f::foo::<alloc::vec::Vec<(alloc::string::String, &[u8; 123])>>`:
+
+```
+thread 'main' panicked at f.rs:2:5:
+explicit panic
+stack backtrace:
+  0: std::panicking::begin_panic
+    at /rustc/d6c...582/library/std/src/panicking.rs:769:5
+  1: f::foo::<alloc::vec::Vec<(alloc::string::String, &[u8; 123])>>
+  2: f::main
+  3: <fn() as core::ops::function::FnOnce<()>>::call_once
+note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
+```
+
+#### Possible drawbacks
+
+Symbols using the v0 mangling scheme can be larger than symbols with the
+legacy mangling scheme, which can result in a slight increase in linking
+times and binary sizes if symbols aren't stripped (which they aren't by default).
+Fortunately this impact should be minor, especially with modern linkers like
+lld, which Rust [will now default to on some targets][switch-to-lld].
+
+Some old versions of tools/distros or niche tools that the compiler team are
+unaware of may not have had support for the v0 mangling scheme added. When
+using these tools, the only consequence is that users may encounter mangled
+symbols. [rustfilt] can be used to demangle Rust symbols if a tool does not.
+
+In any case, using the new mangling scheme can be disabled if any problem
+occurs: use the `-Csymbol-mangling-version=legacy -Zunstable-options` flag
+to revert to using the legacy mangling scheme.
+
+#### Adding v0 support in your tools
+
+If you maintain a tool that interacts with Rust symbols and does not
+support the v0 mangling scheme, there are Rust and C implementations
+of a v0 symbol demangler available in the [rust-lang/rustc-demangle]
+repository that can be integrated into your project.
+
+#### Summary
+
+rustc will use our "v0" mangling scheme on nightly for all targets
+starting in tomorrow's rustup nightly (`nightly-2025-XX-XX`).
+
+Let us know if you encounter problems, by [opening an
+issue](https://github.com/rust-lang/rust/issues/new/choose) on GitHub.
+
+If that happens, you can use the legacy mangling scheme with
+the `-Csymbol-mangling-version=legacy -Zunstable-options` flag.
+Using the legacy mangling scheme requires nightly, it is not intended
```

Comment:
```suggestion
Explicitly setting the legacy mangling scheme requires nightly, it is not intended
```

Everyone on stable is using it, they just can't set it explicitly. This clarification seems important, so people understand how they're using it on stable already.